### ISSUE TITLE
Migrating from helm chart v1.5.5 to v1.12.2

### ISSUE THREAD

**Issue by rolljee** [goharbor/harbor-helm#1540]
Hello I am having a hard time trying a way to migrate from v1.5.5 to v1.12.2

Witch bump all images from v2.1.x to v2.8.x

# What's the problem ?

It's seems like the process need to migrate my data but there are some issues.

1. Harbor template names have changed they go from `harbor-harbor-<service>` to `harbor-service` witch create a new persistent volume claim

This is my pvc that hold my data
<img width="324" alt="Screenshot 2023-07-13 at 08 00 05" src="https://github.com/goharbor/harbor-helm/assets/10114634/ca54481f-b8f9-44f6-a1a5-dd4cb8391918">

The new pvc created is by upgrading is `database-data-harbor-database-0` witch lead to a situation where I believe harbor is taking it as a fresh install
![Screenshot 2023-07-12 at 15 04 21 (1)](https://github.com/goharbor/harbor-helm/assets/10114634/eb9ec98e-79cd-445b-bbf9-e31d5db4928b)

## What I tried ?

I tried to set the persistent volume to an existingClaim, with this `persistence.persistentVolumeClaim.database.existingClaim ` but it does not work, harbor is not taking my previous volume for the database.

I have tried to keep my current version 1.5.5 but upgrading all images to v2.8.x witch keep the same service name but migration end up in a dirty state.

enableMigrateHelmHook does not exist in helm chart version 1.5.5

I have tried to use this property with the helm version v1.12.2 but the process fails looking for a config map that does not exists since my current version of harbor have the name `harbor-harbor-core` not `harbor-core`
<img width="714" alt="Screenshot 2023-07-13 at 08 10 18" src="https://github.com/goharbor/harbor-helm/assets/10114634/6caee1c3-275b-494f-9be8-f3037f40f26b">

I am feeling stuck here, if someone have an idea I'd would gladly appreciate

**Comment by zyyw** [comment]
Hi @rolljee Harbor doesn't support upgrading from v2.1.x to v2.8.x directly in one step. Users need to upgrade within 2 minor versions. For example, you may have to upgrade from v2.1.x to v2.3.x, then to 2.5.x, then to 2.7.x, then 2.8.x.
Please try this upgrade path.

**Comment by rolljee** [comment]
Alight I will, I'll keep you posted on how it goes

**Comment by rolljee** [comment]
Just a follow up on those who need to update as well.

Migration Harbor


1.5.5 -> 1.7.0

Now add -> fullnameOverride: harbor-harbor
Working migration

1.7.0 -> 1.9.0

Now add -> enableMigrateHelmHook: true
Working migration

1.9.0 -> 1.11.0

Working migration

1.11.0 -> 1.12.0

Working migration

Migration is complicated but it works :thumbsup:

**Comment by angelsantillana94** [comment]
Hi @zyyw @rolljee ,

Thank you for the detailed insights shared by the community on upgrading Harbor. However, I noticed that this information (such as the need for intermediate upgrades and specific Helm chart settings like `fullnameOverride` or `enableMigrateHelmHook`) isn't included in the official Harbor documentation:

- Official upgrade docs: https://goharbor.io/docs/1.10/administration/upgrade/helm-upgrade/
- Helm upgrade guide: https://github.com/goharbor/harbor-helm/blob/main/docs/Upgrade.md

This creates challenges for users trying to plan upgrades from their current versions. For example, in my case:

- **App version:** 2.6.2
- **Helm chart version:** 1.10.2
- **Target versions:** Latest app (2.12.0) and Helm chart (1.16.0)

Given the current versions available (`helm search repo harbor/harbor -l`), it’s unclear which intermediate steps I need to follow for a successful upgrade. Based on the existing discussions, I assume the path might involve these steps:

1. Upgrade from `1.10.2 -> 1.12.x` (App `2.6.x -> 2.8.x`)
2.  Then `1.12.x -> 1.14.x` (App `2.8.x -> 2.10.x`)
3. Finally, `1.14.x -> 1.16.0` (App `2.10.x -> 2.12.0`)

However, there’s no official guidance on whether any of these steps require enabling the `enableMigrateHelmHook` option or applying other specific settings like `fullnameOverride`.

Could the documentation be updated to reflect these practices and clarify:

1. The exact intermediate steps required to upgrade between versions?
2. Whether options like enableMigrateHelmHook or others need to be activated at specific stages?

This information would greatly simplify the upgrade process and prevent potential issues for users. Additionally, if there are any known pitfalls specific to my current versions (`2.6.2 / 1.10.2`), please share guidance on how to address them.

Looking forward to your feedback. This will greatly benefit the community and ensure smoother upgrades.

Thank you! 🙏

**Comment by angelsantillana94** [comment]
Ah, I forgot to mention that my issue stems from attempting to upgrade directly from **v2.6.2 to v2.11.1**. After this upgrade, the data no longer appears in the UI.

It seems the database was upgraded from **v13 to v15**, but the data didn’t migrate correctly during this process. Could this be related to skipping intermediate versions or missing configuration settings like **enableMigrateHelmHook**?

Any advice on how to recover the data or properly address this situation would be greatly appreciated.

Thank you in advance for your help!

**Comment by rolljee** [comment]
Hi @angelsantillana94 I think it's hard to keep an up to date documentation since all kubernetes installation can differ in various ways.

we upgraded to helm chart version 1.15 (1 month ago) from 1.12 and we took the path to upgrade the chart one version at a time in order to process the data migration properly.

https://github.com/goharbor/harbor-helm/blob/cccbc3f3c287e06c62271245dc7ae03c8f46753b/README.md?plain=1#L191. 

We removed the enableMigrateHelmHook from the chart and it still worked properly.
I think it's because we now use a managed postgres database in our infrastructure for harbor.

So in the end I think the only best way to update is doing one version at a time. Because postgres itself does not migrate properly when jumping 2 majors in one go.

That's why in the documentation they say it is important to backup the database before launching an upgrade
